### DISCUSSION TITLE
Native SVG icon support

### DISCUSSION THREAD

**Discussion by Garulf** [Flow-Launcher/Flow.Launcher#652] [Ideas]
My plugin [Search-MDI](https://github.com/Garulf/Search-MDI) allows users to search [materialdesignicons.com](https://materialdesignicons.com/)'s very large library of icons. However Flow doesn't preview `SVG` images, and shows the default program that handles the image file.

Microsoft's [PowerToys](https://github.com/microsoft/PowerToys) allows Explorer to handle `SVG` which allows Flow Launcher as well. Would this be an easy addon?

**Comment by taooceros** [comment]
So you mean with power toy, the svg will be displayed without any change in Flow?

**Reply by Garulf** [reply]
Yes

**Reply by taooceros** [reply]
That's really interesting. Will take a look on that.
Besides that, we may have another option to handle the svg with drawableimage in wpf. Though probably that will make the result a bit complex.

How do you handle svg? Download it and transfer it to flow?

**Reply by Garulf** [reply]
SVG files are really XML files with coordinates of the image. For example:
```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">
<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" version="1.1"  width="24" height="24" viewBox="0 0 24 24">
   <path fill="#000000" d="M3.9 4L9 10.03C7.58 10.17 6.36 11.18 6 12.59L4 20H6.07L7.92 13.11C8.09 12.46 8.69 12 9.36 12H10.69L17.47 20H20.1L15 13.97C16.42 13.83 17.64 12.82 18 11.41L20 4H17.93L16.08 10.89C15.91 11.54 15.31 12 14.64 12H13.31L6.53 4Z" />
</svg>
```

Im reading the database and writing an XML file on the fly while pointing Flow to the file path.

**Reply by Garulf** [reply]
I could convert the SVG to a rasterized image but there are no good libraries that handle this very well, not to mention its performance penalties.

Alternative is to download the png from the website but that's even slower.

**Reply by taooceros** [reply]
We can support svg. But let's see whether we can merge svg control with the image control, or else there will be three controls (and the glyph) that tries to do the same thing.

**Comment by pc223** [comment]
+1 this, I think `svg` will be a highly used in the future.

I noticed this bug when try out the `Search-MDI` plugin, Flow fails to display `svg` image, **and** it **automatically** switch to thumbnail (which on my computer is MS Edge). I think this is a bit of a weird behavior, implicitly switch to thumbnail of a image I mean :smile:

@taooceros What do you think about this behavior? Should it consider a bug?

**Reply by taooceros** [reply]
The behavior is under expectations. Display image takes too much memory. Probably svg may not contain this kind of memory issue.
